src/pybind/ceph_argparse.py

```diff
@@ -531,13 +531,13 @@ def valid(self, s, partial=False):
         else:
             t, i = s.split('.', 1)
             if t not in ('osd', 'mon', 'client', 'mds', 'mgr'):
-                raise ArgumentValid('unknown type ' + t)
+                raise JsonFormat('unknown type ' + t)
```

Comment:
in addition to the outcome of the fix, please explain the reason *why* this change fixes the symptom you were seeing in the commit message. and a test case would be great.

Reply:
per the docstring of `JsonFormat` 

> some syntactic or semantic issue with the JSON

but this is not a JSON related error. so i am inclined not using `JsonFormat` here.

Reply:
Yes, I'm left confused by this as well. Perhaps `ArgumentValid` was caught up the stack and this `JsonFormat` avoids that? This looks hacky.

Reply:
Yup! I went through the code again and re-debugged each flows, where I found that when `find_cmd_target()` is trying to validate the command by calling `CephName` and `CephPgid`, it is failing in the `valid()` of `CephPgid` where it is trying to access a variable `poolid` in the except clause, which isn't assigned yet due to error there. Now, the code works as expected and we are getting the proper error message for `wrong cephname type` i.e. `unknown type ....`